writers/csv_report_writer.rb

```diff
@@ -0,0 +1,16 @@
+require 'csv'
+class CsvReportWriter
+  def write(data)
+    CSV.open("report.csv", "w") do |csv|
+      csv << ["file name", "GPS data"] # headers
+
+      data.each do |file_name, gps|
+        csv << [file_name, gps]
+      end
+    end
+  end
+
+  def self.write(data)
+    CsvReportWriter.new.write(data)
```

Comment:
awesome, I love that you did this! One nitpick: class/static methods are usually located above instance methods, so I would move this to the top of the file. Ruby doesn't care about this but it's just convention :) 

Another small tidbit that you might find interesting is ruby's [`...` operator](https://blog.saeloun.com/2019/12/04/ruby-2-7-adds-new-operator-for-arguments-forwarding), which is used for forwarding method arguments. Using that, you could rewrite this as 

```rb
def self.write(...)
  new(...).write
end
```

or even more concisely, using [endless methods](https://www.bigbinary.com/blog/ruby-3-adds-endless-method-definition)


```rb
def self.write(...) = new(...).write
```

Reply:
Good callout. Updated